### ISSUE TITLE
About Results

### ISSUE THREAD

**Issue by XiaoYangon** [ZongyuGuo/Inpainting_FRRN#5]
hello，I use the pretrained model to predict places2 val data，but there are some noises in some image results such as: left is groundtruth,right is output of model
![下载](https://user-images.githubusercontent.com/40583872/75762400-9294bf00-5d75-11ea-8da7-f7d6e0794f8e.png)
![下载 (20)](https://user-images.githubusercontent.com/40583872/75762736-2797b800-5d76-11ea-98d2-fbeb086f40b5.png)
![下载 (21)](https://user-images.githubusercontent.com/40583872/75762764-2f575c80-5d76-11ea-9358-cb8b40175d32.png)

Is it the same result in your experiment？

**Comment by ZongyuGuo** [comment]
Well. Actually the first example shows so-called 'checkerboard' artifact, which occasionally happens when mask is large. This phenomenon is mentioned by several papers on learning-based inpainting. I have two suggestions to alleviate this problem:
1. During training, you can increase the weight of GAN loss to generate subjectively better results. Additionally, you can replace our GAN network to stronger one such as the network used in GatedConv.
2. Inconsistence of dataset distribution may cause this problem as well.

As for the second and the third example, such noise did not appear in my local environment. Does this phenomenon happen frequently? How about quantitative results?

**Comment by XiaoYangon** [comment]
Thank you very much for your reply. Yes. there are many results like the second and the third example. I haven't modified any code. What is your environment configuration? Is it convenient to see your test code.

**Comment by ZongyuGuo** [comment]
`img = Image.open('image.png').convert("RGB")
mask = Image.open('mask.png').convert("L")
img = np.array(img).transpose((2, 0, 1))[None, ...]
mask = np.array(mask)[None, None, ...]
img = torch.Tensor(img).float().cuda() / 255.0
mask = (torch.Tensor(mask)>0).float().cuda()
outputs, gen_loss, dis_loss = inpaint_model.process(img, 1-mask)
img_out = outputs[0].detach().permute(1, 2, 0).cpu().numpy() * 255.0
img_out = Image.fromarray(img_out.astype(np.uint8))
img_out.save('./image_rec.png')
`

Just like it. A conventional image IO process. Can you send these two images with their masks to me, I can try to generate the result on my local environment.

**Comment by ZongyuGuo** [comment]
`img = Image.open('image.png').convert("RGB")
mask = Image.open('mask.png').convert("L")
img = np.array(img).transpose((2, 0, 1))[None, ...]
mask = np.array(mask)[None, None, ...]
img = torch.Tensor(img).float().cuda() / 255.0
mask = (torch.Tensor(mask)>0).float().cuda()
outputs, gen_loss, dis_loss = inpaint_model.process(img, 1-mask)
img_out = outputs[0].detach().permute(1, 2, 0).cpu().numpy() * 255.0
img_out = Image.fromarray(img_out.astype(np.uint8))
img_out.save('./image_rec.png')`

Just like it. A conventional image IO process. Can you send these two images with their masks to me, I can try to generate the result on my local environment.

**Comment by ZongyuGuo** [comment]
Sorry. I directly copy this code from another _issue_.

![image](https://user-images.githubusercontent.com/28563876/75786282-00ee7700-5da0-11ea-984b-db0e33d46666.png)

**Comment by XiaoYangon** [comment]
> `img = Image.open('image.png').convert("RGB") mask = Image.open('mask.png').convert("L") img = np.array(img).transpose((2, 0, 1))[None, ...] mask = np.array(mask)[None, None, ...] img = torch.Tensor(img).float().cuda() / 255.0 mask = (torch.Tensor(mask)>0).float().cuda() outputs, gen_loss, dis_loss = inpaint_model.process(img, 1-mask) img_out = outputs[0].detach().permute(1, 2, 0).cpu().numpy() * 255.0 img_out = Image.fromarray(img_out.astype(np.uint8)) img_out.save('./image_rec.png')`
> 
> Just like it. A conventional image IO process. Can you send these two images with their masks to me, I can try to generate the result on my local environment.

hello, these images are from places2 val data(256*256),the masks  are  from Irregular mask in the paper Image Inpainting for Irregular Holes Using Partial Convolutions. They're all random results. I can't give them to you. but you should have them ,can you predict val data?

**Comment by ZongyuGuo** [comment]
I try to select some similar mask from Irregular Mask dataset on these two images. They are completed with noise as well. 
![img_save](https://user-images.githubusercontent.com/28563876/75841181-0a132e80-5e08-11ea-84c6-bb8bc2cbe430.png)

I guess there may be three reasons.
1. Actually this released model is not a mature version. After re-organizing the code, we only check its quantitative performance and do not check the subjective quality. Longer training time may help.
2. Normalization layer may be a reason. _Region Normalization for Image Inpainting_ This paper reveals the normalization problem existing in inpainting domain. So when you train your own model using our code, you can through away all normalization layer.
3. Our model has an upper bound for the hole size. You can stack more FRR-Blocks to enhance the network ability.

**Comment by XiaoYangon** [comment]
Thank you very much for your explanation. I'll try your solution.
